### DISCUSSION TITLE
save() as image works but not as PDF

### DISCUSSION THREAD

**Discussion by colinmackinlay** [spatie/browsershot#480] [Q&A]
`Browsershot::url('https://bbc.co.uk')->save('test.png');` works fine as does `test.jpg`
But
`Browsershot::url('https://bbc.co.uk')->save('test.pdf');`
 crashes with error below.

Feels like a Windows issue but seems strange that it's only the pdf rendering that is affected. Any ideas anyone?

```
The command "node ^"C:^\Users^\Colin^\Websites^\CSS^\vendor^\spatie^\browsershot^\src/../bin/browser.js^" ^"^{^\^"url^\^":^\^"https:^\/^\/bbc.co.uk^\^",^\^"action^\^":^\^"pdf^\^",^\^"options^\^":^{^\^"path^\^":^\^"test.pdf^\^",^\^"args^\^":^[^],^\^"viewport^\^":^{^\^"width^\^":800,^\^"height^\^":600^}^}^}^"" failed. Exit Code: 1(General error) Working directory: C:\Users\Colin\Websites\CSS\public Output: ================ Error Output: ================ Error: Protocol error (IO.read): Read failed at C:\Users\Colin\Websites\CSS\node_modules\puppeteer\lib\cjs\puppeteer\common\Connection.js:208:63 at new Promise (<anonymous>) at CDPSession.send (C:\Users\Colin\Websites\CSS\node_modules\puppeteer\lib\cjs\puppeteer\common\Connection.js:207:16) at Object.readProtocolStream (C:\Users\Colin\Websites\CSS\node_modules\puppeteer\lib\cjs\puppeteer\common\helper.js:249:39) at async Page.pdf (C:\Users\Colin\Websites\CSS\node_modules\puppeteer\lib\cjs\puppeteer\common\Page.js:1187:16) at async getOutput (C:\Users\Colin\Websites\CSS\vendor\spatie\browsershot\bin\browser.js:34:14) at async callChrome (C:\Users\Colin\Websites\CSS\vendor\spatie\browsershot\bin\browser.js:252:18)
```

**Comment by colinmackinlay** [comment]
Removing puppeteer in project and installing globally (and making sure to run command as administrator) stopped this error but now returns 
```
Spatie\Browsershot\Exceptions\CouldNotTakeBrowsershot
For some reason Chrome did not write a file at `c:\test.png`.
```
for
`Browsershot::url('https://bbc.co.uk')->save('test.png');`
and also when doing pdf or jpg

Must be another weird windows permission issue

**Reply by pgrimaud** [reply]
> Must be another weird windows permission issue

I guess too. Did you try to save the picture to your current path instead of the root of your computer? 

```php
Browsershot::url('https://bbc.co.uk')->save(dirname(__FILE__) . '/test.png');
```

**Reply by colinmackinlay** [reply]
> > Must be another weird windows permission issue
> 
> I guess too. Did you try to save the picture to your current path instead of the root of your computer?
> 
> ```
> Browsershot::url('https://bbc.co.uk')->save(dirname(__FILE__) . '/test.png');
> ```

Thanks for the thought. All that does is try to save elsewhere and still errors:
`For some reason Chrome did not write a file at 'C:\Users\Colin\Websites\CSS\routes/test.png.'`
